src/Symfony/Component/DependencyInjection/Compiler/ResolveDefinitionTemplatesPass.php

```diff
@@ -127,7 +127,8 @@ private function resolveDefinition($id, DefinitionDecorator $definition)
                 throw new RuntimeException(sprintf('Invalid argument key "%s" found.', $k));
             }
 
-            $index = (int) substr($k, strlen('index_'));
+            // strlen('index_') = 6
+            $index = (int) substr($k, 6);
```

Comment:
This is a run once compiler pass, I think readability is more important than optim here